### ISSUE TITLE
Encountered known unsupported method torch.randn/torch.Tensor.get_device/torch.Tensor.cuda

### ISSUE THREAD

**Issue by kshamaramesh** [NVIDIA-AI-IOT/torch2trt#192]
Hi ,

I am currently finding the following issues while converting to torch2trt.

Please find the code snippet or model below
`class VAEGen(nn.Module):
    # VAE architecture
    def __init__(self, input_dim):
        super(VAEGen, self).__init__()
        # dim = params['dim']
        # n_downsample = params['n_downsample']
        # n_res = params['n_res']
        # activ = params['activ']
        # pad_type = params['pad_type']

        # content encoder
        self.enc = ContentEncoder(2, 4, 3, 64, 'in', 'relu', pad_type='reflect')
        self.dec = Decoder(2, 4, self.enc.output_dim, 3, res_norm='in', activ='relu', pad_type='reflect')

    def forward(self, images):
        # This is a reduced VAE implementation where we assume the outputs are multivariate Gaussian distribution with mean = hiddens and std_dev = all ones.
        hiddens,_ = self.encode(images)
        if self.training == False:
            noise = Variable(torch.randn(hiddens.size()).cuda(hiddens.data.get_device()))
            images_recon = self.decode(hiddens + noise)
        else:
            images_recon = self.decode(hiddens)
        return images_recon, hiddens

    def encode(self, images):
        hiddens = self.enc(images)
        noise = Variable(torch.randn(hiddens.size()).cuda(hiddens.data.get_device()))
        return hiddens, noise

    def decode(self, hiddens):
        images = self.dec(hiddens)
        return images`

**Comment by jaybdub** [comment]
Hi kshamarmesh.

Thanks for reaching out!

This model may be difficult to convert because of the randomly generated tensor.   I would recommend either.

### Option 1 - Optimize only Encoder / Decoder

```python
model.enc = torch2trt(...)
model.dec = torch2trt(...)
```

### Option 2 - Refactor to provide noise as input

```python
def forward(self, images, noise):
    # This is a reduced VAE implementation where we assume the outputs are multivariate Gaussian distribution with mean = hiddens and std_dev = all ones.
    hiddens,_ = self.encode(images)
    if self.training == False:
        images_recon = self.decode(hiddens + noise)
    else:
        images_recon = self.decode(hiddens)
    return images_recon, hiddens
```

Please let me know if this helps or you have any questions.

Best,
John
